### PR TITLE
feat: add GetCodeownersErrors to RepositoriesService (#2405)

### PR DESCRIPTION
**Pull request by cbrgm**
This PR adds a new function `GetCodeownersErrors` to the `RepositoriesService` struct.
It adds support for the endpoint https://docs.github.com/en/rest/repos/repos#list-codeowners-errors as described in #2405

### CODE REVIEW

File: github/repos_codeowners.go
Comment (gmlewis) [inline github/repos_codeowners.go:1]: Please add the standard copyright header with the year 2022 here before the `package` line with a blank line separating the two.

File: github/repos_codeowners.go
Comment (gmlewis) [inline github/repos_codeowners.go:19]:
```suggestion
	Suggestion *string `json:"suggestion,omitempty"`
```

File: github/repos_codeowners.go
Comment (gmlewis) [inline github/repos_codeowners.go:14]: Looking at the official docs for this data, it appears that all the fields will be returned except the `suggestion` field, so I'm thinking that we might want to break from the surrounding practice in this repo and make the rest of these fields values instead of pointers.

File: github/repos_codeowners_test.go
Comment (gmlewis) [inline github/repos_codeowners_test.go:1]: Same here, please.

**Review by gmlewis — CHANGES_REQUESTED** [review]
This is looking great, @cbrgm - thank you!
Just a few minor tweaks, please, before we have a second reviewer approval.

File: github/repos_codeowners.go
Comment (cbrgm) [reply github/repos_codeowners.go:6]: Done! Thanks, forgot about that.

File: github/repos_codeowners_test.go
Comment (cbrgm) [reply github/repos_codeowners_test.go:6]: 👍

File: github/repos_codeowners.go
Comment (cbrgm) [reply github/repos_codeowners.go:19]: Changed that. Do you think
```go
// CodeownersErrors represents a list of syntax errors detected in the CODEOWNERS file.
type CodeownersErrors struct {
	Errors []*CodeownersError `json:"errors"`
}
```
should also be a slice of values instead of a slice of pointers?

Comment (gmlewis) [reply github/repos_codeowners.go:19]: No, I think that is fine because when we have slices of structs, we typically use pointers to reduce the chance of errors when iterating over the items... so let's leave this one.  Thanks!

**Review by gmlewis — APPROVED** [review]
Thank you, @cbrgm !
LGTM.

Awaiting LGTM+Approval from any other contributor to this repo before merging.

**Review by spikeb4 — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @spikeb4 !
Merging.

**Pull request merged**
